File: RecyclerViewWithCursorLoader/app/src/main/java/com/nex3z/examples/recyclerviewwithcursorloader/ContactAdapter.java
package com.nex3z.examples.recyclerviewwithcursorloader;

import android.content.Context;
import android.database.Cursor;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class ContactAdapter extends  RecyclerView.Adapter<ContactAdapter.ViewHolder> {
    private Cursor mCursor;

    @Override
    public ContactAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        View view = inflater.inflate(R.layout.item, parent, false);

        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(ContactAdapter.ViewHolder holder, int position) {
        mCursor.moveToPosition(position);

        String name = mCursor.getString(MainActivityFragment.COL_DISPLAY_NAME);
        holder.mContact.setText(name);
    }

    @Override
    public int getItemCount() {
        if (mCursor != null) {
            return mCursor.getCount();
        } else {
            return 0;
        }
    }

    public void swapCursor(Cursor newCursor) {
        mCursor = newCursor;
        notifyDataSetChanged();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView mContact;

        public ViewHolder(View itemView) {
            super(itemView);
            mContact = (TextView) itemView.findViewById(R.id.contact);
        }
    }
}

File: ArchitectureComponents/ArchitectureComponents/app/src/main/java/com/nex3z/examples/architecturecomponents/data/ContactAdapter.kt
package com.nex3z.examples.architecturecomponents.data

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.nex3z.examples.architecturecomponents.R
import com.nex3z.examples.architecturecomponents.data.entity.Contact
import kotlinx.android.synthetic.main.item_contact.view.*

class ContactAdapter : RecyclerView.Adapter<ContactAdapter.ViewHolder>() {

    var contacts: List<Contact> = emptyList()
    var onItemClickListener: OnItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent?.context)
        val itemView = inflater.inflate(R.layout.item_contact, parent, false)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
        if (holder == null) {
            return
        }
        
        val contact = contacts[position]
        holder.itemView.tv_contact_name.text = contact.name
    }

    override fun getItemCount(): Int = contacts.size

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        init {
            itemView.setOnClickListener { onItemClickListener?.onItemClick(adapterPosition) }
        }
    }

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }
}